plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergNodeLocalDynamicSplitPruning.java

```diff
@@ -187,6 +195,151 @@ public void testDynamicSplitPruningOnUnpartitionedTable()
+
+        IcebergConfig icebergConfig = new IcebergConfig();
+        HiveTransactionHandle transaction = new HiveTransactionHandle(false);
+        try (TempFile file = new TempFile()) {
+            Files.delete(file.path());
+
+            TrinoOutputFile outputFile = new LocalOutputFile(file.file());
+            TrinoInputFile inputFile = new LocalInputFile(file.file());
+            List<String> columnNames = ImmutableList.of(dateColumnName, receiptColumnName, amountColumnName);
+            List<Type> types = ImmutableList.of(DATE, VARCHAR, amountColumnType);
+
+            try (OrcWriter writer = new OrcWriter(
+                    OutputStreamOrcDataSink.create(outputFile),
+                    columnNames,
+                    types,
+                    toOrcType(tableSchema),
+                    NONE,
+                    new OrcWriterOptions(),
+                    ImmutableMap.of(),
+                    true,
+                    OrcWriteValidation.OrcWriteValidationMode.BOTH,
+                    new OrcWriterStats())) {
+                BlockBuilder dateBuilder = DATE.createBlockBuilder(null, 1);
+                DATE.writeLong(dateBuilder, dateColumnValue);
+                BlockBuilder receiptBuilder = VARCHAR.createBlockBuilder(null, 1);
+                VARCHAR.writeString(receiptBuilder, receiptColumnValue);
+                BlockBuilder amountBuilder = amountColumnType.createBlockBuilder(null, 1);
+                writeShortDecimal(amountBuilder, amountColumnValue.unscaledValue().longValueExact());
+                writer.write(new Page(dateBuilder.build(), receiptBuilder.build(), amountBuilder.build()));
+            }
+
+            IcebergSplit split = new IcebergSplit(
+                    inputFile.toString(),
+                    0,
+                    inputFile.length(),
+                    inputFile.length(),
+                    -1, // invalid; normally known
+                    ORC,
+                    PartitionSpecParser.toJson(partitionSpec),
+                    PartitionData.toJson(new PartitionData(new Object[] {dateColumnValue})),
+                    ImmutableList.of(),
+                    SplitWeight.standard());
+
+            String tablePath = inputFile.location().fileName();
+            TableHandle tableHandle = new TableHandle(
+                    TEST_CATALOG_HANDLE,
+                    new IcebergTableHandle(
+                            CatalogHandle.fromId("iceberg:NORMAL:v12345"),
+                            "test_schema",
+                            tableName,
+                            TableType.DATA,
+                            Optional.empty(),
+                            SchemaParser.toJson(tableSchema),
+                            Optional.of(PartitionSpecParser.toJson(partitionSpec)),
+                            2,
+                            TupleDomain.withColumnDomains(ImmutableMap.of(dateColumnHandle, Domain.singleValue(DATE, dateColumnValue))),
```

Comment:
Extract as variable `unenforcedDataPredicate` and use it instead of `sameDatePredicate`